### PR TITLE
Dev/1.3.0

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Comment by TomKimsey** [comment]
Code formatting is failing :(

**Review by TomKimsey — CHANGES_REQUESTED** [review]
Fix code formatting before review

File: src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java [hunk 1]
```diff
@@ -552,59 +539,39 @@ public static boolean rerunProvisioning() {
...
+      provisioningMqttMgr.startWithExponentialRetry(
+          AutomaticRetryCodeExponential.MAX_RETRIES_UNLIMITED_VALUE,
+          SCTimeUnit.SECONDS.toMillis(CConnectorProvisionMqttMgr.MAX_SECONDS_BETWEEN_RETRIES));
+      Logger.LOG_DEBUG("Finished creating and starting device provisioning MQTT manager.");
+
+      provisioningMqttMgr.awaitProvisioning();
+      provisioningMqttMgr.stop();
+      Logger.LOG_DEBUG("Finished device provisioning MQTT.");
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Failed to run MQTT device provisioning due to an Exception!");
```
Comment (TomKimsey) [inline src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java:574]:
```suggestion
      Logger.LOG_CRITICAL("Failed to run MQTT device provisioning due to an exception!");
```

```suggestion
      Logger.LOG_CRITICAL("Exception occurred, failed to run MQTT device provisioning!");
```

File: src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java [hunk 2]
```diff
@@ -617,56 +584,33 @@ protected void codeToRetry() {
...
+              connectorConfig.getStringUtf8Support(),
+              mqttPort,
+              mqttRootCaFilePath,
+              mqttUsername,
+              mqttPassword);
+      mqttMgr.startWithExponentialRetry(
+          AutomaticRetryCodeExponential.MAX_RETRIES_UNLIMITED_VALUE,
+          SCTimeUnit.SECONDS.toMillis(CConnectorMqttMgr.MAX_SECONDS_BETWEEN_RETRIES));
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL("Failed to run MQTT setup due to an Exception!");
```
Comment (TomKimsey) [inline src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java:613]:
```suggestion
      Logger.LOG_CRITICAL("Failed to run MQTT setup due to an exception!");
```

```suggestion
      Logger.LOG_CRITICAL("Exception occurred, failed to run MQTT setup!");
```

File: src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java [hunk 3]
```diff
@@ -680,15 +624,21 @@ protected void codeToRetry() {
...
     if (mqttMgr != null) {
-      DefaultEventHandler.delMqttListener(mqttMgr);
-      mqttMgr.disconnect();
-      mqttMgr.stopMqttThread();
-      mqttMgr = null;
+      try {
+        mqttMgr.restart();
+      } catch (Exception e) {
+        restartSuccess = false;
+        Logger.LOG_SERIOUS("An error occurred while restarting the MQTT manager!");
```
Comment (TomKimsey) [inline src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java:633]: Why are other uses calling out exception, where as this is calling out error?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
 # Ewon Flexy Cumulocity Connector Changelog
 
+## Version 1.3.0
+### Features
+- Added option to change Cumulocity server certificate in configuration file
+- Update MQTT to use new library ConstrainedMqttManager which simplifies and improves error handling
```
Comment (TomKimsey) [inline CHANGELOG.md:6]:
```suggestion
- Update MQTT to use new library ConstrainedMqttManager.java which simplifies and improves error handling
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
+### Bug Fixes
+- N/A
+### Other
+- Updated extensions library to 1.12.0
```
Comment (TomKimsey) [inline CHANGELOG.md:10]: Would this be more clear?
```suggestion
- Updated com.hms_networks.americas.sc:extensions library to version 1.12.0
```

File: README.md [hunk 6]
```diff
@@ -130,6 +130,14 @@ running the connector.
 
 Optional string parameter to specify the port of the Cumulocity server. Default: "8883"
 
+#### Certificate URL (CertificateUrl)
+
+Optional string parameter to specify the URL of the Cumulocity server's certificate. 
+Default: "https://certs.godaddy.com/repository/gdroot-g2.crt"
```
Comment (TomKimsey) [inline README.md:136]: Is this something hosted by Cumulocity or are we hosting something on godaddy?

File: README.md [hunk 7]
```diff
@@ -130,6 +130,14 @@ running the connector.
...
+
+This parameter may be useful when connecting to Cumulocity Edge servers which use a self-signed
```
Comment (TomKimsey) [inline README.md:138]: Is there ever a situation where it is not useful when connecting to an edge server using a self-signed cert? If so you could remove the "may" as its very passive.

**Review by TomKimsey — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (alexjhawk) [reply README.md:136]: This is the certificate used by Cumulocity cloud, and it is publicly downloadable from GoDaddy itself.

File: README.md (hunk 7 above)
Comment (alexjhawk) [reply README.md]: Reworded

File: CHANGELOG.md (hunk 5 above)
Comment (alexjhawk) [reply CHANGELOG.md]: Yes, I agree with this change. Updated

File: CHANGELOG.md (hunk 4 above)
Comment (alexjhawk) [reply CHANGELOG.md]: changed

File: src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java (hunk 3 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/sc/cumulocity/CConnectorMain.java]: changed

**Review by TomKimsey — APPROVED** [review]

**Review by oliver-walker-hms — APPROVED** [review]

**Pull request merged**